crates/turbopack-ecmascript/src/tree_shake/pkg_json/mod.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::{bail, Result};
+use serde::Deserialize;
+use turbo_tasks::Vc;
+use turbo_tasks_fs::{FileContent, FileSystemPath};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct PackageJson {
+    #[serde(default = "true_by_default")]
+    side_effects: bool,
+}
+
+fn true_by_default() -> bool {
+    true
+}
+
+#[turbo_tasks::function]
+pub async fn is_side_effect_free(filename: Vc<FileSystemPath>) -> Result<Vc<bool>> {
+    let package_json = read_package_json_for(filename.clone()).await?;
```

Comment:
You probably want to use the existing `read_package_json` that we already have

Reply:
Thanks! I don't understand how I missed it.